Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -198,4 +201,147 @@ public override RestRequest BuildRequest(RestRequest request)
+        {
+            this.BucketNotificationConfiguration = config;
+            return this;
+        }
+    }
+
+    public class RemoveAllBucketNotificationsArgs : BucketArgs<RemoveAllBucketNotificationsArgs>
+    {
+        public RemoveAllBucketNotificationsArgs()
+        {
+            this.RequestMethod = Method.PUT;
+        }
+        public override RestRequest BuildRequest(RestRequest request)
+        {
+            request.AddQueryParameter("notification","");
+            BucketNotification bucketNotificationConfiguration = new BucketNotification();
+            string body = utils.MarshalXML(bucketNotificationConfiguration, "http://s3.amazonaws.com/doc/2006-03-01/");
+            request.AddParameter(new Parameter("text/xml", body, ParameterType.RequestBody));
+
+            return request;
+        }
+    }
+
+    public class ListenBucketNotificationsArgs : BucketArgs<ListenBucketNotificationsArgs>
+    {
+        internal string Prefix { get; private set; }
+        internal string Suffix { get; private set; }
+        internal List<EventType> Events { get; private set; }
+        internal IObserver<MinioNotificationRaw> NotificationObserver { get; private set; }
+        public bool EnableTrace { get; private set; }
+
+        public ListenBucketNotificationsArgs()
+        {
+            this.RequestMethod = Method.GET;
+            this.EnableTrace = false;
+            this.Events = new List<EventType>();
+            this.Prefix="";
+            this.Suffix="";
+        }
+
+        public ListenBucketNotificationsArgs WithNotificationObserver(IObserver<MinioNotificationRaw> obs)
+        {
+            this.NotificationObserver = obs;
+            return this;
+        }
+        public override RestRequest BuildRequest(RestRequest request)
+        {
+            request.AddQueryParameter("prefix",this.Prefix);
+            request.AddQueryParameter("suffix",this.Suffix);
+            foreach (var eventType in this.Events)
+            {
+                request.AddQueryParameter("events",eventType.value);
+            }
+            request.ResponseWriter = responseStream =>
+            {
+                using (responseStream)
+                {
+                    var sr = new StreamReader(responseStream);
+                    while (true)
+                    {
+                        string line = sr.ReadLine();
+                        if (this.EnableTrace)
+                        {
+                            Console.WriteLine("== ListenBucketNotificationsAsync read line ==");
+                            Console.WriteLine(line);
+                            Console.WriteLine("==============================================");
+                        }
+                        if (line == null)
+                        {
+                            break;
+                        }
+                        string trimmed = line.Trim();
+                        if (trimmed.Length > 2)
```

Comment:
what does this `trimmed.Length` check filter out?

Reply:
To filter out empty stream lines with just newline whitespaces.